mlrun/feature_store/retrieval/job.py

```diff
@@ -62,9 +62,14 @@ def run_merge_job(
     function = run_config.to_function(kind, merger.get_default_image(kind))
 
     # Avoid overriding a handler that was provided by the user
-    # The user shouldn't have to provide a handler, but we leave this option open just in case
     if not run_config.handler:
         function.with_code(body=default_code)
+    else:
+        # The user shouldn't have to provide a handler
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            "When passing `run_config` to `get_offline_features`, specifying"
+            " the `handler` parameter of the `run_config` is not allowed."
```

Comment:
I think this is much shorter and to the point
```suggestion
            "get_offline_features does not support run_config with a handler"
```